xla/service/algebraic_simplifier.cc

```diff
@@ -4626,11 +4626,39 @@ Status AlgebraicSimplifierVisitor::HandleBroadcast(HloInstruction* broadcast) {
   return OkStatus();
 }
 
+HloInstruction* MakeNewCompare(HloInstruction* compare, HloInstruction* a,
+                               HloInstruction* b) {
+  return compare->AddInstruction(HloInstruction::CreateCompare(
+      compare->shape(), a, b, compare->comparison_direction()));
+}
+
 Status AlgebraicSimplifierVisitor::HandleCompare(HloInstruction* compare) {
   HloInstruction* lhs;
   HloInstruction* rhs;
   CHECK(Match(compare, m::Compare(m::Op(&lhs), m::Op(&rhs))));
 
+  // Gt(Max(a,b), a) -> Gt(b,a)
+  // Gt(Max(a,b), b) -> Gt(a,b)
+  // Gt(a, Min(a,b)) -> Gt(a,b)
+  // Gt(b, Min(a,b)) -> Gt(b,a)
+  if (compare->comparison_direction() == ComparisonDirection::kGt) {
+    HloInstruction* a;
+    HloInstruction* b;
+    if (Match(lhs, m::Maximum(m::Op(&a), m::Op(&b)))) {
+      if (rhs == a) {  // Gt(Max(a,b), a) -> Gt(b,a)
+        return ReplaceInstruction(compare, MakeNewCompare(compare, b, a));
```

Comment:
I think you can also just use MakeCompareHlo from hlo_creation_utils.h instead of the helper method:

```
MakeCompareHlo(ComparsionDirection::kGt, b, a, &compare->metadata())
```

Alternatively if you prefer your helper method, please make it pass the metadata to AddInstruction.

Reply:
switched to `MakeCompareHlo` from `hlo_creation_utils.h`